envoy/grpc/status.h

```diff
@@ -52,6 +54,17 @@ class Status {
     // was invalid.
     InvalidCode = -1,
   };
+
+  /**
+   * The gRPC status option for gRPC local reply.
+   */
+  struct LocalReplyGrpcStatusOption {
```

Comment:
Maybe move this struct definition to the header file defining "sendLocalReply".

Reply:
Other than `envoy/http/filter.h`, I also changed `envoy/http/codec.h` so both of them need to access its definition.

Reply:
Hi Alyssa, it would be nice to keep this PR alone for only the refactor and https://github.com/envoyproxy/envoy/pull/20620#discussion_r842742436 adds the logic for gRPC error details. Or you mean only adding the new detail field? but it still needs  https://github.com/envoyproxy/envoy/pull/20620#discussion_r842742436 to enable it. WDYT?

> if we want to have a generic struct or gRPC struct as long as we're refactoring here?

You mean a protobuf struct, why do you consider using that?

Reply:
+1 to just having `LocalReplyOptions` as I'm sure as Alyssa says someone will come along want to add non-gRPC options and that will make the future refactor not needed. So let's just move this to a new header file somewhere if we have circular dependency issues?